pkg/reconciler/clusterducktype/testdata/features/smoke.feature

```diff
@@ -0,0 +1,21 @@
+Feature: Reconcile ClusterDuckType
+
+    Scenario Outline: Reconciling Smoke Test with generation <generation>.
+
+        Given the following objects (from file):
+            | file                      | generation   |
+            | config/smoke/initial.yaml | <generation> |
+
+        And a ClusterDuckType reconciler
+
+        When reconciling "zhangchas.duck.knative.dev"
+
+        Then expect status updates (from file):
+            | file                      | generation   |
+            | config/smoke/updated.yaml | <generation> |
```

Comment:
Neat!

Reply:
How flexible is this? Can you say "Given this .. Then that ... Then Delete this .. then that ..." etc.. ?

Reply:
all things after file in the headers define how the `{{ .name }}` will work, so here, generation is not defined in my test code, but is in the yaml.

Reply:
Actually my question was more about Cucumber itself. I got my answer by looking at it. This is really cool!